contracts/dual_token/DualStakingTokenActivityChecker.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+// DualStakingToken interface
+interface IDualStakingToken {
+    function mapMutisigs(address multisig) external view returns (bool);
+}
+
+// Multisig interface
+interface IMultisig {
+    function nonce() external view returns (uint256);
+}
+
+/// @dev Zero address.
+error ZeroAddress();
+
+/// @dev Zero value.
+error ZeroValue();
+
+/// @dev Unauthorized account.
+/// @param account Account address.
+error UnauthorizedAccount(address account);
+
+/// @dev State is locked.
+error Locked();
+
+/// @title DualStakingTokenActivityChecker - Smart contract for performing dual token service staking activity check
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract DualStakingTokenActivityChecker {
```

Comment:
This is just for demo purposes? Wouldn't we just fall back to whatever activity checker the underlying OLAS staking uses?

Reply:
We need a locked activity checker, if you are asking about that difference in the code. Otherwise all other functions follow whatever logic is in the required activity checker. I'll use the inheritance to show how this works.

Reply:
Some logic specific to the second token might also be needed